daffodil-runtime1/src/main/scala/org/apache/daffodil/processors/DataProcessor.scala

```diff
@@ -357,7 +357,14 @@ class DataProcessor private (
 
   def save(output: DFDL.Output): Unit = {
 
-    val oos = new ObjectOutputStream(new GZIPOutputStream(Channels.newOutputStream(output)))
+    val os = Channels.newOutputStream(output)
+
+    // write a null-terminated ASCII string as a simple version identifier
+    val headerString = "DAFFODIL " + Misc.getDaffodilVersion + "\u0000"
+    os.write(headerString.getBytes("US-ASCII"))
```

Comment:
I reviewed how we call `getBytes` in Daffodil in order to check for inconsistencies and best practices.  I noticed two things: 1) we call `getBytes("ascii")` instead every other place where we want bytes from ASCII characters; and 2) we call `getBytes` without a charset name too many times.  Java's platform default charset is specific to the user and OS.  On many modern Linux systems, it's UTF-8. On Macs, it’s MacRoman. In the US on Windows, it's often CP1250, while in Europe it's CP1252 or in China it's often simplified Chinese (Big5 or a GB*).  I'm agnostic whether we use "ascii", "US-ASCII", or import java.nio.charset.StandardCharsets and use StandardCharsets.US_ASCII (I see Daffodil typically uses all-lowercase strings most often to save space and typing), but we probably should create a bug to replace all parameter-less `getBytes` calls with `getBytes("utf-8")`.

Reply:
Agreed. It looks like most uses (maybe all?) of `getBytes` without an encoding are in tests, so I"m not sure there's a bug in Daffodil proper. But it does mean our tests are possibly fragile to the system default encoding which we definitely don't want. And improving is definitely worth it.

I've created [DAFFODIL-2561](https://issues.apache.org/jira/browse/DAFFODIL-2561) for this. I'll also update this to use "utf-8" encoding for conistency.